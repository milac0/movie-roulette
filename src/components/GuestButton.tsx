import React, { useContext } from "react";
import { css } from "emotion";
// import { primary } from "../theme/theme";
import axios from "axios";
import { UserContext } from "../context/UserContext";
import { colors } from "./../theme/theme";

const btn = css`
  background: ${colors.secondary};
  border: none;
  border-radius: 5px;
  font-size: 1rem;
  padding: 0.5em 1em;
  &:hover {
    background: ${colors.secondaryDark};
  }
`;

interface Props {}

const GuestButton: React.FC<Props> = () => {
  const { setUser } = useContext(UserContext);
  const handleClick = async () => {
    const response = await axios.get(
      `/authentication/guest_session/new?api_key=${process.env.API_KEY}`
    );
    setUser(response.data);
  };
  return (
    <button className={btn} onClick={handleClick}>
      Guest
    </button>
  );
};

export default GuestButton;
