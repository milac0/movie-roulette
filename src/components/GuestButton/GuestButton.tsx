import React, { useContext, useState } from "react";
import axios from "axios";
import { UserContext } from "../../context/UserContext";
import styles from "./guestButton.scss";

interface Props {}

const GuestButton: React.FC<Props> = () => {
  const { setUser, isAuth } = useContext(UserContext);
  const [expand, setExpand] = useState(false);

  const handleLogin = async () => {
    setExpand(false);
    const response = (
      await axios.get(
        `/authentication/guest_session/new?api_key=${process.env.API_KEY}`
      )
    ).data;
    setUser(response);
    localStorage.setItem("user", JSON.stringify(response));
  };

  const handleLogout = () => {
    localStorage.removeItem("user");
    setUser({
      success: false,
      guest_session_id: "",
      expires_at: ""
    });
  };

  const handleAccountInfo = () => {
    setExpand(!expand);
  };

  return isAuth() ? (
    <div className={styles.account}>
      <p className={styles.accountBtn} onClick={handleAccountInfo}>
        Hello, Guest
      </p>
      {expand && (
        <div className={styles.accountOptions}>
          <p onClick={handleLogout}>Sign out</p>
        </div>
      )}
    </div>
  ) : (
    <button className={styles.btn} onClick={handleLogin}>
      Login as Guest
    </button>
  );
};

export default GuestButton;
