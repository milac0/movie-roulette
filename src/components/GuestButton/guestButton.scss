@import "../../theme/variables";

.btn {
  background: $secondary;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  font-size: 0.75rem;
  outline: none;
  padding: 0.5em 0.5em;
  &:hover {
    background: $secondaryDark;
  }
}

.account {
  margin-right: 0;
  position: relative;

  .accountBtn {
    border: 1px solid #fff;
    border-radius: 5px;
    cursor: pointer;
    font-size: 0.75rem;
    margin: 0;
    margin-bottom: 0.25em;
    outline: none;
    padding: 0.5em;

    &:hover {
      background: #fff;
      color: $primary;
    }
  }

  div {
    background: $backgroundGrey;
    border-radius: 5px;
    color: black;
    position: absolute;
    right: -1.25em;
    width: 140px;
    z-index: 1;

    p {
      border-radius: 5px;
      cursor: pointer;
      margin: 0.25em;
      padding: 0.5em 1em;

      &:hover {
        background: #aaaaaa;
      }
    }
  }
}

@media (min-width: 414px) {
  .account {
    .accountBtn {
      font-size: 1rem;
    }
  }

  .btn {
    font-size: 1rem;
    padding: 0.5em 1em;
  }
}
