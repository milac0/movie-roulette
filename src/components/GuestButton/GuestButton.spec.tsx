import React from 'react';
import renderer from 'react-test-renderer';

import GuestButton from './GuestButton';

describe('GuestButton renders correctly', () => {
    test('snapshot renders', () => {
        const tree = renderer.create(<GuestButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});