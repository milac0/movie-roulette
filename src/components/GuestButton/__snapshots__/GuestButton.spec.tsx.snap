// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestButton renders correctly snapshot renders 1`] = `
<button
  className="btn"
  onClick={[Function]}
>
  Login as Guest
</button>
`;
